nhaystack-rt/src/nhaystack/util/TypeUtil.java

```diff
@@ -226,6 +228,46 @@ else if (def instanceof BSimple)
 
             return simple;
         }
+        else if (def instanceof BEnumOverride)
+        {
+            BEnumOverride cpx = (BEnumOverride) def;
+            BFacets facets = comp.getAction(action.getName()).getFacets();
+            BEnumRange range = (BEnumRange) facets.get(BFacets.RANGE);

```

Comment:
What happens if there is not a BEnumRange configured in the facets for the EnumWritable? Does 'range' get set to null?

Reply:
True, I should guard against wrong comp + action combination here